plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -250,20 +262,41 @@ processSingleBinding = \case
     b -> Just <$> forMOf bindingSubterms b processTerm
 
 maybeAddSubst
-    :: Inlining tyname name uni fun a m
+    :: forall tyname name uni fun a m . Inlining tyname name uni fun a m
     => Strictness
     -> name
     -> Term tyname name uni fun a
     -> m (Maybe (Term tyname name uni fun a))
 maybeAddSubst s n rhs = do
-    -- Only do PostInlineUnconditional
-    -- See Note [Inlining approach and 'Secrets of the GHC Inliner']
-    rhs' <- processTerm rhs
-    doInline <- postInlineUnconditional s rhs'
-    if doInline then do
-        modify (\subst -> extendTerm n (Done rhs') subst)
-        pure Nothing
-    else pure $ Just rhs'
+    -- NOTE:  Nothing means that we are inlining the term:
```

Comment:
This should maybe be a comment on the function, since it's about what the return type means?